src/qtcore/qml/QObject.js

```diff
@@ -26,5 +29,7 @@ function QObject(parent) {
 
         if (this.$parent && this.$parent.$tidyupList)
             this.$parent.$tidyupList.splice(this.$parent.$tidyupList.indexOf(this), 1);
+
+        this.parent = undefined; // must do this. => 1) parent will be notified and erase object from it's children. 2) DOM node will be removed.
```

Comment:
I think here is a better place, because it guaranties that parent property will be cleared. In case of line 19, there is no guaranties because $delete might be called by someone else.